pkg/apis/wksprovider/machine/os/os.go

```diff
@@ -891,13 +892,25 @@ func createFluxSecretFromGitData(gitData GitParams, params SeedNodeParams) ([]by
 
 func wksControllerManifest(controllerImageOverride, namespace, configDir string) ([]byte, error) {
 	var manifestbytes []byte
+
+	// The controller manifest is taken, in order:
+	// 1. from the specified git repository checkout.
+	// 2. from the YAML manifest built-in the binary.
+	//
+	// The controller image is, in priority order:
+	// 1. controllerImageOverride provided on the apply command line.
+	// 2. the image from the manifest if we have found a manifest in the git repository checkout.
+	// 3. quay.io/wksctl/controller:version.ImageTag
 	filepath, err := findControllerManifest(configDir)
 	if err != nil {
 		file, openErr := manifests.Manifests.Open("04_controller.yaml")
 		if openErr != nil {
 			return nil, openErr
 		}
 		manifestbytes, err = ioutil.ReadAll(file)
+		if controllerImageOverride == "" {
+			controllerImageOverride = "quay.io/wksctl/controller:" + version.ImageTag
```

Comment:
this effectively hardwires the version that's set by a linker flag. this would help a lot with maintainability (and reuse of wksctl as source in other projects) if we made it a propagated parameter instead.

Reply:
That value is now threaded from the top level command through the seed parameters.